app/src/main/kotlin/vergecurrency/vergewallet/service/model/DataManager.kt

```diff
@@ -0,0 +1,56 @@
+package vergecurrency.vergewallet.service.model
+
+import android.content.Context
+import io.realm.Realm
+import io.realm.RealmConfiguration
+import vergecurrency.vergewallet.helpers.utils.WalletDataIdentifierUtils
+import java.util.*
+import kotlin.collections.HashMap
+
+object DataManager {
+    var xvgDataRealm: Realm? = null;
+
+
+    fun loadWalletData(context: Context, walletId: UUID) {
+        EncryptedPreferencesManager.getOrCreateEncryptedSharedPreferences(context, walletId)
+        getOrCreateVergeDataRealm(walletId)
+    }
+
+
+    //Return identifier and wallet name of all existing encryptedSharedPreferences
+    fun getAllEncryptedPreferences(context: Context): HashMap<UUID, String> {
+        val existingWalletFiles: HashMap<UUID, String> = HashMap<UUID, String>();
+        val datDir = context.dataDir.listFiles();
+        if (datDir != null) {
+            for (dataDir in datDir) {
```

Comment:
Stream over datdir and use functionalities like map and reduce to get this working. It's barely readable with having 8 if indentations